clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -2507,13 +2441,9 @@ bool SemaHLSL::CheckBuiltinFunctionCall(unsigned BuiltinID, CallExpr *TheCall) {
     break;
   }
   case Builtin::BI__builtin_hlsl_dot: {
-    if (SemaRef.checkArgCount(TheCall, 2))
```

Comment:
Why did you remove this? Since you are leaving it for many of the other builtins this one now looks like the odd man out.

Reply:
BuiltinVectorToScalarMath checks that the arg count is 2. I removed checkArgCount checks in places where the next thing checked was something like BuiltinVectorMath, which checks arg count.

Reply:
a comment to this effect would be good, because this sticks out.

Reply:
Yeah I added a comment